integration/spark/src/main/common/java/io/openlineage/spark/api/AbstractQueryPlanOutputDatasetBuilder.java

```diff
@@ -0,0 +1,21 @@
+package io.openlineage.spark.api;
+
+import io.openlineage.client.OpenLineage;
+import org.apache.spark.scheduler.SparkListenerEvent;
+import org.apache.spark.scheduler.SparkListenerJobEnd;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionEnd;
+
+public abstract class AbstractQueryPlanOutputDatasetBuilder<P extends LogicalPlan>
```

Comment:
Similar to the input dataset builder, please add javadoc explaining the intended use of this class and the fact that only end events will trigger subclasses.